scripts/github-status-report.js

```diff
@@ -9,121 +9,139 @@ const fetch = require('node-fetch')
 const util = require('util')
 const exec = util.promisify(require('child_process').exec)
 
-const GH_API_HEADERS = {
-	'Accept': 'application/vnd.github.v3+json,application/vnd.github.jean-grey-preview+json',
-	'Authorization': `token ${process.env['DANGER_GITHUB_API_TOKEN']}`
+const GH_API_HEADERS = token => {
+  return {
+    Accept:
+      'application/vnd.github.v3+json,application/vnd.github.jean-grey-preview+json',
+    Authorization: `token ${token}`,
+  }
 }
 
-const checkContext = (context) => {
-	if(!context) throw new Error("No context given, failing.")
-	return context
+const checkContext = context => {
+  if (!context) throw new Error('No context given, failing.')
+  return context
 }
 
-const checkStatus = (status) => {
-	if(!status) throw new Error("No status passed, failing.")
-
-	switch(status) {
-		case 'pending':
-		case 'failure':
-		case 'success':
-		case 'error':
-			return status
-			break
-		default:
-			throw new Error(`Invalid status value '${status}'`)
-	}
+const checkStatus = status => {
+  if (!status) throw new Error('No status passed, failing.')
+
+  switch (status) {
+    case 'pending':
+    case 'failure':
+    case 'success':
+    case 'error':
+      return status
+    default:
+      throw new Error(`Invalid status value '${status}'`)
+  }
 }
 
 async function checkToken(token) {
-	// this endpoint does not count against rate limit and has the added benefit of being a preflight check
-	let url = `${GITHUB_API}/rate_limit`
-	let options = {
-		mecthod: 'GET',
-		headers: GH_API_HEADERS
-	}
-
-	let value = await fetch(url, options)
-		.then(response => response.json())
-		.then((response) => {
-			return response
-		})
-
-	return value
+  // this endpoint does not count against rate limit and has the added benefit of being a preflight check
+  let url = `${GITHUB_API}/rate_limit`
+  let options = {
+    mecthod: 'GET',
+    headers: GH_API_HEADERS(token),
+  }
+
+  let value = await fetch(url, options)
+    .then(response => response.json())
+    .then(response => {
+      return response
```

Comment:
Drive-by comment: this `.then` clause serves no purpose.

Reply:
Yeah, yeah, yeah... I know